log/zap.go

```diff
@@ -185,30 +192,30 @@ var Nop = zap.WrapCore(func(zapcore.Core) zapcore.Core {
 // returns the resulting Logger. It's safe to use concurrently.
 func (l Log) WithOptions(opts ...zap.Option) Log {
 	lgr := l.logger.WithOptions(opts...)
-	return Log{logger: lgr}
+	return Log{logger: lgr, name: l.name}
 }
 
 // Info prints message with fields
 func (fl FieldLogger) Info(msg string, fields ...LoggableField) {
-	fl.l.Info(msg, unpack(fields)...)
+	fl.l.Info(msg, unpack(append(fields, String("name", fl.name)))...)
```

Comment:
The problem is, once you add a field, you can't remove it or change it. If we add `name` as a field, then if a child logger is created, it will get two `name` fields, e.g., `{name: p2p, name: p2p.TcpNet}`. This logic is smarter, see the code at: https://github.com/spacemeshos/go-spacemesh/pull/2427/files/c1c336a3c685bef597d5dcab0f83bd729dc13cfa#diff-6e36b5b54c2b1b501474b0cb1be773f051a3710ae1754f16ce1e479b4577c46bR148-R157

Reply:
I added an explanatory comment to this effect

Reply:
Oh, I assumed that fields are entries in a map-like object, that can be overwritten. I just followed the code and realized this is not the case. Counter-intuitive and less useful, if you ask me, but 🤷🏻‍♂️.